src/app/routes/googleSignInRoutes.js

```diff
@@ -58,9 +69,9 @@ module.exports = (app) => {
       keyCloakToken = await createSession(googleProfile.emailId, reqQuery, req, res).catch(handleCreateSessionError);
       logger.info({msg: 'keyCloakToken fetched' + JSON.stringify(keyCloakToken)});
       errType = 'UNHANDLED_ERROR';
-      redirectUrl = reqQuery.redirect_uri.split('?')[0];
+      redirectUrl = reqQuery.redirect_uri.replace(KEYCLOACK_AUTH_CALLBACK_STRING, ''); // to avoid 401 auth errors from keycloak
       if (reqQuery.client_id === 'android') {
-        redirectUrl = redirectUrl + getQueryParams(keyCloakToken);
+        redirectUrl = reqQuery.redirect_uri.split('?')[0] + getQueryParams(keyCloakToken);
```

Comment:
This code will fail in mobile use-case right. When there are UTM params in URL. This has to verify on mobile.